modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormGuestUploadFieldUtilTest.java

```diff
@@ -0,0 +1,278 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.form.web.internal.display.context.util;
+
+import com.liferay.dynamic.data.mapping.model.DDMForm;
+import com.liferay.dynamic.data.mapping.model.DDMFormField;
+import com.liferay.dynamic.data.mapping.model.DDMFormInstance;
+import com.liferay.dynamic.data.mapping.model.DDMFormInstanceRecord;
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.model.impl.DDMStructureImpl;
+import com.liferay.dynamic.data.mapping.service.DDMFormInstanceRecordLocalService;
+import com.liferay.dynamic.data.mapping.test.util.DDMFormTestUtil;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import org.powermock.api.support.membermodification.MemberMatcher;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+/**
+ * @author Carolina Barbosa
+ */
+@RunWith(PowerMockRunner.class)
+public class DDMFormGuestUploadFieldUtilTest {
+
+	@Before
+	public void setUp() throws Exception {
+		setUpDDMForm();
+		setUpDDMFormInstanceRecordLocalService();
+	}
+
+	@Test
+	public void testHasGuestUploadFieldWithNoUploadField() throws Exception {
+		Assert.assertFalse(
+			_ddmFormGuestUploadFieldUtil.hasGuestUploadField(
+				mockDDMFormInstance()));
+	}
+
+	@Test
+	public void testHasGuestUploadFieldWithUploadFieldAllowedForGuests()
+		throws Exception {
+
+		addGuestUploadField();
+
+		Assert.assertTrue(
+			_ddmFormGuestUploadFieldUtil.hasGuestUploadField(
+				mockDDMFormInstance()));
+	}
+
+	@Test
+	public void testHasGuestUploadFieldWithUploadFieldNotAllowedForGuests()
+		throws Exception {
+
+		_ddmForm.addDDMFormField(
+			new DDMFormField("fieldName", "document_library"));
+
+		Assert.assertFalse(
+			_ddmFormGuestUploadFieldUtil.hasGuestUploadField(
+				mockDDMFormInstance()));
+	}
+
+	@Test
+	public void testIsMaxSubmissionLimitReachedWithGuestUserAndUploadFieldNotAllowedForGuests()
+		throws Exception {
+
+		_ddmForm.addDDMFormField(
+			new DDMFormField("fieldName", "document_library"));
```

Comment:
@carolmariaabb, can we reuse the method `addGuestUploadField` here? If we add a parameter to this method, I understand that we can reuse it. What do you think?